subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/deps/ClassDependentsAccumulator.java

```diff
@@ -33,6 +33,7 @@
 public class ClassDependentsAccumulator {
```

Comment:
I understand you've just slight changed this class, but some nitpicking advice would be:

- I don't think the empty default constructor should exist.
- There're too many unnecessary public methods in this class.
- Maybe `Map<String, Set<String>>`  can be replaced with `Multimap` because `parentToChildren` field is already using `Multimap`. This can eliminate `rememberClass` method at least.

Reply:
Good points, I'll take another look at this one.